MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -890,6 +893,54 @@ public void SetVertexBuffer(VertexBuffer vertexBuffer, int vertexOffset)
                                    ? _vertexBuffers.Clear()
                                    : _vertexBuffers.Set(vertexBuffer, vertexOffset);
         }
+
+        private void _SetVertexBuffers(VertexBufferBinding[] vertexBuffers, int bindingCount)
+        {
+            if (vertexBuffers == null || vertexBuffers.Length == 0 || bindingCount == 0)
+            {
+                _vertexBuffersDirty |= _vertexBuffers.Clear();
+            }
+            else
+            {
+                if (vertexBuffers.Length > _maxVertexBufferSlots || bindingCount > _maxVertexBufferSlots)
+                {
+                    var message = string.Format(CultureInfo.InvariantCulture, "Max number of vertex buffers is {0}.", _maxVertexBufferSlots);
+                    throw new ArgumentOutOfRangeException("vertexBuffers", message);
+                }
+
+                _vertexBuffersDirty |= _vertexBuffers.Set(vertexBuffers, bindingCount);

```

Comment:
Actually...  i think you don't need `_SetVertexBuffers` at all here.  This line....

`_vertexBuffersDirty |= _vertexBuffers.Set(vertexBuffers, bindingCount);`

Is the only bit you need.  Might was well put it directly in the overloaded `SetVertexBuffers` calls.